### PR TITLE
IGNITE-12788 Adds cluster fully rebalanced state metric.

### PR DESCRIPTION
**Pull request by ololo3000**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/ClusterRebalancedMetricTest.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        awaitPartitionMapExchange(true, true, null, true);
+
+        assertClusterRebalancedMetricOnAllNodes(exp);
+    }
+
+    /**
+     * Checks that {@link GridMetricManager#CLUSTER_REBALANCED} metric is set to {@code exp} on all cluster nodes.
+     */
+    private void assertClusterRebalancedMetricOnAllNodes(boolean exp) {
+        G.allGrids().stream().allMatch(ignite -> {
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/ClusterRebalancedMetricTest.java]: `allMatch` result ignored.

Comment (ololo3000) [reply modules/core/src/test/java/org/apache/ignite/spi/discovery/ClusterRebalancedMetricTest.java]: Thanks. Done.

**Comment by NSAmelchev** [comment]
@anton-vinogradov , Hi.

Can the new metric be placed at PME registry wihtout creating a new registry?

i.e. `pme.ClusterRebalanced` instead of `rebalance.ClusterRebalanced`

**Comment by anton-vinogradov** [comment]
> @anton-vinogradov , Hi.
> 
> Can the new metric be placed at PME registry wihtout creating a new registry?
> 
> i.e. `pme.ClusterRebalanced` instead of `rebalance.ClusterRebalanced`

**Comment by nizhikov** [comment]
Metrics naming makes sense.

**Pull request merged**